### PR TITLE
add UINavigationBarAppearance implementations

### PR DESCRIPTION
**Pull request by mohammadalijf**
Add `UINavigationBarAppearance` implementations for NavigationController to fix iOS 13 UINavigationBar's barTint and other appearance features

### CODE REVIEW

**Comment by mohammadalijf** [comment]
this PR will fix #5468

**Comment by mohammadalijf** [comment]
Hi @guyca 
CI failed with error below
```
error: Unable to find a destination matching the provided destination specifier:
05:17:38 		{ platform:iOS Simulator, OS:latest, name:iPhone X }
```
I am guessing the agent is using XCode 11 which does not have iPhone X in its simulators.

**Comment by mohammadalijf** [comment]
@guyca 
hi, the CI always gets failed becuase there is no iOS 13 simulator for iPhone X

```xcodebuild: error: Unable to find a destination matching the provided destination specifier:
02:37:27 		{ platform:iOS Simulator, OS:latest, name:iPhone X }
02:37:27 
02:37:27 	The requested device could not be found because no available devices matched the request.
02:37:27 
02:37:27 	Available destinations for the "ReactNativeNavigation" scheme:
02:37:27 		{ platform:macOS, arch:x86_64, variant:Mac Catalyst, id:420ED0EB-89D6-726F-E38F-BB6721BEF480 }
02:37:27 		{ platform:iOS Simulator, id:354D198B-DE6A-4BFD-A8AA-820302A5A82C, OS:12.2, name:iPad (5th generation) }
02:37:27 		{ platform:iOS Simulator, id:0BBD45DE-3F22-4C95-B2B3-C0AA124F447F, OS:12.2, name:iPad (6th generation) }
02:37:27 		{ platform:iOS Simulator, id:9C81BC15-1470-406D-9773-889D426D60A8, OS:12.2, name:iPad Air }
02:37:27 		{ platform:iOS Simulator, id:F5C8FD74-285E-4470-8D32-D9198C2C35FB, OS:12.2, name:iPad Air (3rd generation) }
02:37:27 		{ platform:iOS Simulator, id:F80347C2-34F9-45A9-9C32-EF7135C6D136, OS:13.0, name:iPad Air (3rd generation) }
02:37:27 		{ platform:iOS Simulator, id:5850B0EF-F100-4643-A7B8-D5A722431871, OS:12.2, name:iPad Air 2 }
02:37:27 		{ platform:iOS Simulator, id:FE9DB1AF-7D7E-4149-A42F-7E577211A1EF, OS:12.2, name:iPad Pro (9.7-inch) }
02:37:27 		{ platform:iOS Simulator, id:53648E7D-6162-4885-8F44-5153986AD2A1, OS:13.0, name:iPad Pro (9.7-inch) }
02:37:27 		{ platform:iOS Simulator, id:0FA410A8-CABD-476E-8CE2-1EE19352E6C1, OS:12.2, name:iPad Pro (10.5-inch) }
02:37:27 		{ platform:iOS Simulator, id:9D173944-5B6B-4C1F-8322-DEC7598C761B, OS:12.2, name:iPad Pro (11-inch) }
02:37:27 		{ platform:iOS Simulator, id:71EE2321-48BD-4745-9899-E39BAC58A9D0, OS:13.0, name:iPad Pro (11-inch) }
02:37:27 		{ platform:iOS Simulator, id:3570FBBB-6872-4866-BA05-8E0A18022EB5, OS:12.2, name:iPad Pro (12.9-inch) }
02:37:27 		{ platform:iOS Simulator, id:418A3ED5-C056-40E6-95AF-3E4FB23371E2, OS:12.2, name:iPad Pro (12.9-inch) (2nd generation) }
02:37:27 		{ platform:iOS Simulator, id:17995917-B46C-4F71-9522-3288C0860A08, OS:12.2, name:iPad Pro (12.9-inch) (3rd generation) }
02:37:27 		{ platform:iOS Simulator, id:75AFB261-4858-40C2-9F29-12ACF56BD654, OS:13.0, name:iPad Pro (12.9-inch) (3rd generation) }
02:37:27 		{ platform:iOS Simulator, id:161A3297-044C-4222-BC8B-78C34522B23A, OS:12.2, name:iPhone 5s }
02:37:27 		{ platform:iOS Simulator, id:2FB78817-D527-464A-A31B-FE2705EF9ABF, OS:12.2, name:iPhone 6 }
02:37:27 		{ platform:iOS Simulator, id:C2C77032-D422-4F0E-BA41-B2E77DE0BB13, OS:12.2, name:iPhone 6 Plus }
02:37:27 		{ platform:iOS Simulator, id:3EC61C3D-E7C7-44E3-90CA-DED05F733218, OS:12.2, name:iPhone 6s }
02:37:27 		{ platform:iOS Simulator, id:BC876B52-6654-4CC6-96E7-371FE83819B6, OS:12.2, name:iPhone 6s Plus }
02:37:27 		{ platform:iOS Simulator, id:F7FF5080-286A-4F35-A3CF-483F62AEF791, OS:12.2, name:iPhone 7 }
02:37:27 		{ platform:iOS Simulator, id:46F88D9F-2A31-4BE4-8B69-8E7F3079B157, OS:12.2, name:iPhone 7 Plus }
02:37:27 		{ platform:iOS Simulator, id:96E744B7-1CD9-4998-AD9B-B53AB2BDEA72, OS:12.2, name:iPhone 8 }
02:37:27 		{ platform:iOS Simulator, id:6F2984F8-30E5-41C3-A29B-B207D8D990F3, OS:13.0, name:iPhone 8 }
02:37:27 		{ platform:iOS Simulator, id:340062FA-F423-46A9-94DA-7C1926C588F6, OS:12.2, name:iPhone 8 Plus }
02:37:27 		{ platform:iOS Simulator, id:7FBE6C47-6194-45F4-AEC1-267336EA949F, OS:13.0, name:iPhone 8 Plus }
02:37:27 		{ platform:iOS Simulator, id:0CED83B3-5F90-45E0-B615-3159573F1121, OS:13.0, name:iPhone 11 }
02:37:27 		{ platform:iOS Simulator, id:74A3FA9E-DF85-4DC9-8F18-9EEE1CB0AD55, OS:13.0, name:iPhone 11 Pro }
02:37:27 		{ platform:iOS Simulator, id:AF211BF8-87CB-44A1-89AE-9A416EB4CB12, OS:13.0, name:iPhone 11 Pro Max }
02:37:27 		{ platform:iOS Simulator, id:17AEBEAB-C393-429D-81A9-AC2DD3B8476E, OS:12.2, name:iPhone SE }
02:37:27 		{ platform:iOS Simulator, id:258A3C6B-EFAC-4A4D-8B14-E7F1FD8E9D16, OS:12.2, name:iPhone X }
02:37:27 		{ platform:iOS Simulator, id:FA0DEF47-E702-4163-94EE-AF00F604E90F, OS:12.2, name:iPhone Xs }
02:37:27 		{ platform:iOS Simulator, id:1A2F0195-DB2C-445A-87ED-474A71CA3938, OS:12.2, name:iPhone Xs Max }
02:37:27 		{ platform:iOS Simulator, id:20A18422-E6AF-4A3E-8C34-AE9C84BFDD39, OS:12.2, name:iPhone Xʀ }```
can you fix this. also can you please review this pr?

File: lib/ios/RNNStackController.m [hunk 1]
```diff
@@ -5,6 +5,28 @@
 
 @implementation RNNStackController
 
+- (UINavigationBarAppearance*)getNavigaitonBarStandardAppearance  API_AVAILABLE(ios(13.0)) {
```
Comment (jaredegan) [inline lib/ios/RNNStackController.m]: "navigation" is misspelled several times in this PR as `Navigaiton`

**Comment by guyca** [comment]
Thanks a lot for the PR @mohammadalijf 
This is now fixed in v4

**Pull request closed without merging**